src/dowel/csv_output.py

```diff
@@ -41,13 +41,33 @@ def record(self, data, prefix=''):
                     extrasaction='ignore')
                 self._writer.writeheader()
 
+            # new fieldnames detected, need to read in current file, modify, and write everything again
             if to_csv.keys() != self._fieldnames:
-                self._warn('Inconsistent TabularInput keys detected. '
-                           'CsvOutput keys: {}. '
-                           'TabularInput keys: {}. '
-                           'Did you change key sets after your first '
-                           'logger.log(TabularInput)?'.format(
-                               set(self._fieldnames), set(to_csv.keys())))
+                # obtain extra fieldname
+                extra_field = (set(to_csv.keys()) ^ set(self._fieldnames)).pop()
+
+                # close original log file
+                self._log_file.close()
+
+                # store original data in RAM
+                reader = csv.DictReader(open(self._log_file.name, 'r'))
+                origin_data = []
+                for row in reader:
+                    origin_data.append(row)
```

Comment:
Reading the data like this is definitely a good way to ensure that there's no overwrite mistakes.
A neat tip is that you can actually just use the list function to do this:
```python
    origin_data = list(csv.DictReader(open(self._log_file.name, 'r')))
```